spec/models/miq_expression_spec.rb

```diff
@@ -350,6 +350,112 @@
+      actual = described_class.new("!" => {"=" => {"field" => "Vm-name", "value" => "foo"}}).to_ruby
+      expected = "!(<value ref=vm, type=string>/virtual/name</value> == \"foo\")"
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for an IS NULL expression" do
+      actual = described_class.new("IS NULL" => {"field" => "Vm-name"}).to_ruby
+      expected = "<value ref=vm, type=string>/virtual/name</value> == \"\""
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for an IS NOT NULL expression" do
+      actual = described_class.new("IS NOT NULL" => {"field" => "Vm-name"}).to_ruby
+      expected = "<value ref=vm, type=string>/virtual/name</value> != \"\""
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for an IS EMPTY expression" do
+      actual = described_class.new("IS EMPTY" => {"field" => "Vm-name"}).to_ruby
+      expected = "<value ref=vm, type=string>/virtual/name</value> == \"\""
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for an IS NOT EMPTY expression" do
+      actual = described_class.new("IS NOT EMPTY" => {"field" => "Vm-name"}).to_ruby
+      expected = "<value ref=vm, type=string>/virtual/name</value> != \"\""
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for a FIND expression with checkall" do
+      actual = described_class.new(
+        "FIND" => {"search" => {"=" => {"field" => "Vm-name", "value" => "foo"}},
+                   "checkall" => {">" => {"field" => "Vm.hardware.cpu_sockets", "value" => "2"}}}
+      ).to_ruby
+      expected = "<find><search><value ref=vm, type=string>/virtual/name</value> == \"foo\"</search><check mode=all><value ref=vm, type=string>/virtual/hardware/cpu_sockets</value> > \"2\"</check></find>"
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for a FIND expression with checkany" do
+      actual = described_class.new(
+        "FIND" => {"search" => {"=" => {"field" => "Vm-name", "value" => "foo"}},
+                   "checkany" => {">" => {"field" => "Vm.hardware.cpu_sockets", "value" => "2"}}}
+      ).to_ruby
+      expected = "<find><search><value ref=vm, type=string>/virtual/name</value> == \"foo\"</search><check mode=any><value ref=vm, type=string>/virtual/hardware/cpu_sockets</value> > \"2\"</check></find>"
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for a FIND expression with checkcount" do
+      actual = described_class.new(
+        "FIND" => {"search" => {"=" => {"field" => "Vm-name", "value" => "foo"}},
+                   "checkcount" => {">" => {"field" => "Vm.hardware.cpu_sockets", "value" => "2"}}}
+      ).to_ruby
+      expected = "<find><search><value ref=vm, type=string>/virtual/name</value> == \"foo\"</search><check mode=count><count> > 2</check></find>"
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for a KEY EXISTS expression" do
```

Comment:
@gtanzillo similarly, not sure I'm using these KEY/VALUE EXISTS expressions correctly